Venmo/src/main/java/com/braintreepayments/api/VenmoClient.java

```diff
@@ -428,6 +447,219 @@ private Intent getLaunchIntent(Configuration configuration, String profileId, St
         return venmoIntent;
     }
 
+    /**
+     * Use this method with the manual browser switch integration pattern.
+     *
+     * @param browserSwitchResult a {@link BrowserSwitchResult} with a {@link BrowserSwitchStatus}
+     * @param callback            {@link VenmoOnActivityResultCallback}
+     */
+    public void onBrowserSwitchResult(@NonNull BrowserSwitchResult browserSwitchResult, @NonNull final VenmoOnActivityResultCallback callback) {
+        int result = browserSwitchResult.getStatus();
+        switch (result) {
+            case BrowserSwitchStatus.CANCELED:
+                braintreeClient.sendAnalyticsEvent("pay-with-venmo.app-links.canceled");
+                callback.onResult(null, new UserCanceledException("User canceled Venmo."));
+                break;
+            case BrowserSwitchStatus.SUCCESS:
+                Uri deepLinkUri = browserSwitchResult.getDeepLinkUrl();
+                if (deepLinkUri != null) {
+                    if (deepLinkUri.getPath().contains("success")) {
+                        String resourceId = parseResourceId(String.valueOf(deepLinkUri));
+                        String paymentMethodNonce = parsePaymentMethodNonce(String.valueOf(deepLinkUri));
+                        String username = parseUsername(String.valueOf(deepLinkUri));
+
+                        if (resourceId != null) {
+                            venmoApi.createNonceFromPaymentContext(resourceId, new VenmoOnActivityResultCallback() {
+
+                                @Override
+                                public void onResult(@Nullable VenmoAccountNonce venmoAccountNonce, @Nullable Exception error) {
+                                    if (venmoAccountNonce != null) {
+                                        braintreeClient.sendAnalyticsEvent("pay-with-venmo.app-links.success");
+                                        callback.onResult(venmoAccountNonce, null);
+                                    } else {
+                                        callback.onResult(null, error);
+                                    }
+                                }
+                            });
+                        } else if (paymentMethodNonce != null && username != null) {
+                            braintreeClient.sendAnalyticsEvent("pay-with-venmo.app-links.success");
+                            VenmoAccountNonce venmoAccountNonce = new VenmoAccountNonce(paymentMethodNonce, username, false);
+                            callback.onResult(venmoAccountNonce, null);
```

Comment:
Should there be vaulting logic here?

Reply:
Ah! You're absolutely right there should be. Made this update to match `onActivityResult`: https://github.com/braintree/braintree_android/pull/911/commits/4b538dc18c54c52203a24521fe0fcb1cc7ccb7a3